doc/source/reference/arrays.indexing.rst

```diff
@@ -180,6 +180,16 @@ concepts to remember include:
    create an axis of length one. :const:`newaxis` is an alias for
    'None', and 'None' can be used in place of this with the same result.
 
+.. note::
+
+    NumPy slicing creates a reference instead of a copy as in the case of
+    oridinary Python objects. Care must be taken when extracting
+    a small portion from a large array which becomes useless after the
+    extraction, because the small portion extracted contains a reference
+    to the large original array whose memory will not be released until
+    all arrays derived from it are garbage-collected. In such cases an
+    explicit ``copy()`` is recommended.
+
```

Comment:
in the first line `reference` -> view, even better 
```
:term:`view`
```
to make it a link to the glossary term.